src/parser_yang.c

```diff
@@ -4516,6 +4519,64 @@ yang_check_deviate(struct lys_module *module, struct unres_schema *unres, struct
 
         /* ... and replace it with the value specified in deviation */
         dev_target->flags |= deviate->flags & LYS_CONFIG_MASK;
+
+        /* "config" is explicitely set in the node */
+        dev_target->flags |= LYS_CONFIG_SET;
+
+        /* "config" must be set exclusively to "yes" or "no" */
+        assert(!((dev_target->flags & LYS_CONFIG_W) && (dev_target->flags & LYS_CONFIG_R)));
+
+        /* from rfc7950#page-86: All key leafs in a list MUST have the same value
+         * for their "config" as the list itself.
+         */
+        if ((node_list = lys_is_key((const struct lys_node_leaf *)dev_target, NULL))) {
+            if ((node_list->flags & LYS_CONFIG_MASK) != (dev_target->flags & LYS_CONFIG_MASK)) {
+                LOGVAL(module->ctx, LYE_INSTMT, LY_VLOG_NONE, NULL, "config");
+                LOGVAL(module->ctx, LYE_SPEC, LY_VLOG_NONE, NULL, "The deviate causes a violation of \"config\" in key leaf.");
```

Comment:
I suppose you wanted to keep the message short so fine, but some details could be included. But at least make it `The deviate causes a violation of \"config\" rules in a key leaf.` so that it makes more sense. Similarly for the other cases.